### PR TITLE
Minified/Generated code check

### PR DESCRIPTION
**Pull request by boyter**
For https://github.com/boyter/scc/issues/92

Allows you to turn on minified/generated code checks which flags the files differently OR exclude them entirely.

The power is yours so choose wisely!

### CODE REVIEW

File: processor/workers.go [hunk 1]
```diff
@@ -486,6 +488,15 @@ func CountStats(fileJob *FileJob) {
 		if fileJob.Content[index] == '\n' || index >= endPoint {
 			fileJob.Lines++
 
+			if MinifiedGenerated {
+				if avgLineByteCount == 0 {
+					avgLineByteCount = index - lineOffset
+				} else {
+					avgLineByteCount = (avgLineByteCount + (index - lineOffset)) / 2
```
Comment (dbaggerman) [inline processor/workers.go]: This isn't a real average - it's more of a rolling average weighted towards lines at the end of the file.

This doesn't even seem like it's needed. Just stick `avgLineByteCount := len(fileJob.Content) / fileJob.Lines` after the state machine.

File: processor/workers.go [hunk 2]
```diff
@@ -737,6 +767,15 @@ func fileProcessorWorker(input chan *FileJob, output chan *FileJob) {
 					duplicates.mux.Unlock()
 				}
 
+				if IgnoreMinifiedGenerate {
+					if res.Minified {
```
Comment (dbaggerman) [inline processor/workers.go]: Purely style preference, but I'd make this `IgnoreMinifiedGenerate && res.Minified` over the nested condition.

File: processor/workers.go (hunk 1 above)
Comment (boyter) [reply processor/workers.go]: Ah the old not notice whats going on when you are doing it. Good point.

File: processor/workers.go (hunk 2 above)
Comment (boyter) [reply processor/workers.go]: 100% agree with you. Will refactor.

File: processor/workers.go (hunk 1 above)
Comment (boyter) [reply processor/workers.go]: The weighted towards the end was useful since some JS files follow that pattern but this might still work.

Comment (boyter) [reply processor/workers.go]: Tests pass so all good.

**Comment by boyter** [comment]
If you feel its all good @dbaggerman just merge it. Otherwise ill clean up whatever you throw back asap.

**Comment by dbaggerman** [comment]
Looks good to me.

**Pull request merged**